front/components/assistant/conversation/WebsearchAction.tsx

```diff
@@ -1,16 +1,139 @@
-import type { WebsearchActionType } from "@dust-tt/types";
+import {
+  ChevronDownIcon,
+  ChevronRightIcon,
+  Chip,
+  Icon,
+  MagnifyingGlassStrokeIcon,
+  Spinner,
+  Tooltip,
+} from "@dust-tt/sparkle";
+import type { WebsearchActionType, WebsearchResultType } from "@dust-tt/types";
+import { Transition } from "@headlessui/react";
+import { ExternalLinkIcon } from "lucide-react";
+import { useState } from "react";
+
+import { trimText } from "@app/lib/utils";
 
-// TODO(pr,websearch) Implement this function
 export default function WebsearchAction({
   websearchAction,
 }: {
   websearchAction: WebsearchActionType;
 }) {
+  const { query } = websearchAction;
+
+  const [resultsListVisible, setResultsListVisible] = useState(false);
+
+  const { results } = websearchAction.output || { results: [] };
+  console;
+
   return (
     <>
-      <div>
-        Action to be implemented here: {JSON.stringify(websearchAction)}
+      <div className="flex flex-row items-center gap-2 pb-2">
+        <div className="text-xs font-bold text-element-600">
+          Searching&nbsp;Google&nbsp;for:
+        </div>
+        <Chip.List isWrapping={true}>
+          <Tooltip label={`Query used for google search: ${query}`}>
+            <Chip color="slate" label={trimText(query)} />
+          </Tooltip>
+        </Chip.List>
+      </div>
+      <div className="grid grid-cols-[auto,1fr] gap-2">
+        <div className="grid-cols-auto grid items-center">
+          {!websearchAction.output ? (
+            <div>
+              <div className="pb-2 text-xs font-bold text-element-600">
+                Searching...
+              </div>
+              <Spinner size="sm" />
+            </div>
+          ) : (
+            <div className="text-xs font-bold text-element-600">
+              <span>Results:</span>
+            </div>
+          )}
+        </div>
+        <div className="row-span-1 select-none">
+          {websearchAction.output && (
+            <div
+              onClick={() => setResultsListVisible(!resultsListVisible)}
+              className="cursor-pointer"
+            >
+              <Chip color="purple">
+                {results.length > 0
+                  ? SearchResultsInfo(results)
+                  : "No results found"}
+                <Icon
+                  visual={
+                    resultsListVisible ? ChevronDownIcon : ChevronRightIcon
+                  }
+                  size="xs"
+                />
+              </Chip>
+            </div>
+          )}
+        </div>
+        <div className="col-start-2 row-span-1">
+          {!!results.length && (
+            <Transition
```

Comment:
`nit`

Pretty sure you can achieve the same with className="animate-in fade-in animate-out fade-out duration-200"

Reply:
Cargo Culting m'a tuer